WordPress/Classes/Services/MediaImportService.swift

```diff
@@ -188,16 +188,15 @@ class MediaImportService: NSObject {
         // HEIC isn't supported when uploading an image, so we filter it out (http://git.io/JJAae)
         allowedFileTypes.remove("heic")
 
-        let completion: (Result<Media, Error>) -> Void = { result in
+        let completion: (Error?) -> Void = { error in
             self.coreDataStack.performAndSave({ context in
-                let media = try context.existingObject(with: media.objectID) as! Media
-                switch result {
-                case let .success(media):
```

Comment:
This `media` variable here overwrites the one above, which leads to the updating code below used an media instance bound to other context.

This issue can be caught by turning on the Core Data concurrency launch arguments and uploading a photo to Media Library.

Reply:
Okay. And that's the reason why the completion type went from `Result<Media, Error>` to `Optional<Error>`, right? Do not pass around a `Media` instance and keep using the one bound to execution context instead.

Reply:
Yep, the updated `import` function is a private function. We don't need to make it return an object that is not going to be used.